metrics/squad/README.md

```diff
@@ -0,0 +1,82 @@
+results = squad_metric.compute(predictions=predictions, references=references)
+```
+## Output values
+
+This metric outputs a dictionary with two values: the average exact match score and the average [F1 score](https://huggingface.co/metrics/f1).
+
+    {'exact_match': 100.0, 'f1': 100.0}
+
+The range of `exact_match` is 0-100, where 0.0 means no answers were matched and 100.0 means all answers were matched. 
+
+The range of `f1` is 0-1 -- its lowest possible value is 0, if either the precision or the recall is 0, and its highest possible value is 1.0, which means perfect precision and recall.
+
+## Values from popular papers
+The [original SQuAD paper](https://nlp.stanford.edu/pubs/rajpurkar2016squad.pdf) reported an F1 score of 51.0% and an Exact Match score of 40.0%. They also report that human performance on the dataset represents an F1 score of 90.5% and an Exact Match score of 80.3%.
+
+For more recent model performance, see the [dataset leaderboard](https://paperswithcode.com/dataset/squad).
+
+## Examples 
+
+Maximal values for both exact match and F1 (perfect match):
+
+    from datasets import load_metric
+    squad_metric = load_metric("squad")
+    predictions = [{'prediction_text': '1976', 'id': '56e10a3be3433e1400422b22'}]
+    references = [{'answers': {'answer_start': [97], 'text': ['1976']}, 'id': '56e10a3be3433e1400422b22'}]
+    results = squad_metric.compute(predictions=predictions, references=references)
+    results
+    {'exact_match': 100.0, 'f1': 100.0}
+
+Minimal values for both exact match and F1 (no match):
+
+    from datasets import load_metric
+    squad_metric = load_metric("squad")
+    predictions = [{'prediction_text': '1999', 'id': '56e10a3be3433e1400422b22'}]
+    references = [{'answers': {'answer_start': [97], 'text': ['1976']}, 'id': '56e10a3be3433e1400422b22'}]
+    results = squad_metric.compute(predictions=predictions, references=references)
+    results
+    {'exact_match': 0.0, 'f1': 0.0}
+    
+Partial match (2 out of 3 answers correct) : 
+
+    from datasets import load_metric
+    squad_metric = load_metric("squad")
+    predictions = [{'prediction_text': '1976', 'id': '56e10a3be3433e1400422b22'}, {'prediction_text': 'Beyonce', 'id': '56d2051ce7d4791d0090260b'},  {'prediction_text': 'climate change', 'id': '5733b5344776f419006610e1'}]
+    references = [{'answers': {'answer_start': [97], 'text': ['1976']}, 'id': '56e10a3be3433e1400422b22'}, {'answers': {'answer_start': [233], 'text': ['Beyoncé and Bruno Mars']}, 'id': '56d2051ce7d4791d0090260b'}, {'answers': {'answer_start': [891], 'text': ['climate change']}, 'id': '5733b5344776f419006610e1'}]
+    results = squad_metric.compute(predictions=predictions, references=references)
+    results
+    {'exact_match': 66.66666666666667, 'f1': 66.66666666666667}
+
+## Limitations and bias
+This metric works only with the [SQuAD v.1 dataset](https://huggingface.co/datasets/squad) -- it will not work with any other dataset formats.
```

Comment:
Isn't it more accurate to say that this metric only works for datasets that have the same schema / format as SQuAD v1?

Reply:
well, it's technically a dataset-specific metric, cause there are ids that are internal to SQuAD, but you're right that they don't get checked, really.